core/server/api_container/server/startosis_engine/startosis_interpreter.go

```diff
@@ -262,7 +264,7 @@ func (interpreter *StartosisInterpreter) Interpret(
 	if mainFuncParamsNum >= minimumParamsRequiredForPlan {
 		firstParamName, _ := mainFunction.Param(planParamIndex)
 		if firstParamName == planParamName {
-			kurtosisPlanInstructions := KurtosisPlanInstructions(packageId, interpreter.serviceNetwork, interpreter.recipeExecutor, interpreter.moduleContentProvider)
+			kurtosisPlanInstructions := KurtosisPlanInstructions(packageId, interpreter.serviceNetwork, interpreter.recipeExecutor, interpreter.moduleContentProvider, packageReplaceOptions)
```

Comment:
maybe not worth worrying about - how does idempotence work with replacements?

Reply:
it should work with idempotence because the`packageReplaceOptions` are persisted in the `kurtosis.yml` file